### DISCUSSION TITLE
TC39 signals

### DISCUSSION THREAD

**Discussion by saolof** [alpinejs/alpine#4179] [4. General]
Alpine has a very neat pluggable reactivity system with setReactivityEngine, and currently uses the Vue reactivity system by default as I understand it.

Has anyone tried plugging in the polyfill of the TC39 signals proposal? Though it looks like that proposal is a bit bare bones and includes basic signals but not any kind of deep reactive store. Also, I am curious about what other backends people have tried

**Comment by ekwoka** [comment]
I've not looked at that specifically.

I have made a Signals based reactivity engine that does satisfy Alpine's needs and works. About 30% faster than vue reactivity, and about 70% smaller.

I've not extensively looked at edge cases, but it does support maps and sets like vue reactivity does.

**Reply by saolof** [reply]
That is actually something I was looking for since Vues reactivity model is fairly large, do you have a link to the repository?

**Reply by ekwoka** [reply]
https://github.com/ekwoka/timber/tree/main/packages%2Freactivity

It is on npm as well

https://www.npmjs.com/package/@timberts/reactivity

Would be interesting for you to try and then check if it causes issues.

You should be able to use it with Alpine.setReactivityEngine to test it out.

Or possibly just do a npm alias for Vue reactivity. I believe it implements all of Vue reactivities exports that Alpine uses.
